### PR TITLE
NMS-14763 Add Priority Executor Classes

### PR DESCRIPTION
**Pull request by rssntn67**
Added classes to manage priority scheduling
 Added LegacyPriorityExecutor that use PriorityQueue
 Added Executable that is a ReadyRunnable class with a priority
 Added Tests

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-14763

### CODE REVIEW

**Comment by rssntn67** [comment]
I created the branch from branch

NMS-14771 | Move Common Adapter Enlinkd classes to Core
-- | --


[NMS-14771](https://issues.opennms.org/browse/NMS-14771)	Move Common Adapter Enlinkd classes to Core

So this PR includes also the changes there. In fact new Classes needs Schedulable.java.

**Review by aramosvizcarra — DISMISSED** [review]
LGTM!

File: core/daemon/src/main/java/org/opennms/netmgt/scheduler/Executable.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2018 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2018 The OpenNMS Group, Inc.
```
Comment (christianpape) [inline core/daemon/src/main/java/org/opennms/netmgt/scheduler/Executable.java:5]:
```suggestion
 * Copyright (C) 2022 The OpenNMS Group, Inc.
 * OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.
```

**Review by christianpape — CHANGES_REQUESTED** [review]
Only a header is wrong.

**Review by christianpape — DISMISSED** [review]
LGTM.

**Review by aramosvizcarra — APPROVED** [review]
LGTM!

**Pull request merged**